runtime/compiler/control/CassandraLogger.cpp

```
@@ -0,0 +1,229 @@
+   snprintf(queryString, 256, "CREATE KEYSPACE IF NOT EXISTS %s  WITH REPLICATION = {'class':'SimpleStrategy','replication_factor':1};", _databaseName); 
+   OCassStatement* statement = Ocass_statement_new(queryString, 0);
+
+   OCassFuture* queryFuture = Ocass_session_execute(_session, statement);
+   Ocass_statement_free(statement);
+   if (Ocass_future_error_code(queryFuture) != 0)
+      {
+      /* Display connection error message */
+      const char* message;
+      size_t messageLength;
+      Ocass_future_error_message(queryFuture, &message, &messageLength);
+      fprintf(stderr, "PersistentLogging - Cassandra Database Keyspace Creation Error: '%.*s'\n", (int)messageLength, message);
+      
+      Ocass_future_free(queryFuture);
+      return false;
+      }
+
+   Ocass_future_free(queryFuture);
+   return true;
+
+}
+bool CassandraLogger::createTable(const char *tableName) 
+   {
+   char queryString[256];
+   snprintf(queryString, 256, "CREATE TABLE IF NOT EXISTS %s.%s (clientID text, methodName text, logContent text, insertionDate date,insertionTime time, primary key (clientID, methodName, insertionDate, insertionTime));", _databaseName, tableName);
+   OCassStatement* statement = Ocass_statement_new(queryString, 0);
+   OCassFuture* queryFuture = Ocass_session_execute(_session, statement);
+   Ocass_statement_free(statement);
+   if (Ocass_future_error_code(queryFuture) != 0)
+      {
+      /* Display connection error message */
+      const char* message;
+      size_t messageLength;
+      Ocass_future_error_message(queryFuture, &message, &messageLength);
+      fprintf(stderr, "Persistent Logging - Cassandra Database Table Creation Error: '%.*s'\n", (int)messageLength, message);
+      
+      Ocass_future_free(queryFuture);
+      return false;
+      }
+
+   Ocass_future_free(queryFuture);
+   return true;
+   }
+bool CassandraLogger::connect()
+   {
+   /* Setup and connect to cluster */
+   _cluster = Ocass_cluster_new();
```

If `_cluster` and `_session` can be NULL, we must add a check and fail fast.